entity-framework/core/internals/internal-service-provider.md

```diff
@@ -0,0 +1,166 @@
+---
+title: The internal service provider - EF Core
+author: ajcvickers
+ms.date: 01/25/2020
+uid: core/internals/internal-service-provider
+---
+
+# The internal service provider
+
+## Introduction
+
+Internally, EF Core is built around a [dependency injection]() container.
+This container exposes an extensible set of services that define the behavior of EF Core.
+Service implementations can come from the application, plug-ins, extensions, database providers, and EF Core itself.
+
+This container is exposed as an [`IServiceProvider`]() and implemented using [Microsoft.Extensions.DependencyInjection]().
+It is therefore known as the _internal service provider_.
+
+## DbContext and the internal service provider
+
+[`DbContext`]() instances are created by the application using:
+* Explicit construction with [`new`]()
+* The application's own dependency injection container
+This container is known as the [_application service provider_]().
+
+The DbContext instance then locates the internal service provider to use.<sup>[(1)](#footnote-1)</sup>
+The service provider is most commonly created automatically when needed.
+It is then [cached and reused]().
+
+An application can instead explicitly configure the internal service provider to use.
+The application must then handle caching.
+
+### Scopes and lifetimes
+
+A new [service provider scope]() is created for each DbContext instance.
+This means that:
+* [Singleton]() services are shared between DbContext instances
+* [Scoped]() services are shared within, but not between, DbContext instances
+
+A DbContext instance typically represents a [session]() or [unit-of-work]().
```

Comment:
Maybe say a word about DI scopes at the beginning of this section before going into DbContext etc.?

Reply:
What I'm hoping here is that I can find links to these concepts in the D.I. documentation. I decided not to try to describe D.I. in general here. Depending what I find to link to I will revisit this.